### ISSUE TITLE
mysql permission error, not able to open caravel

### ISSUE THREAD

**Issue by neevany** [apache/incubator-superset#1411]
used master [branch](https://github.com/airbnb/caravel/commit/63161b11c347d5a6d62f7ae7dc91fa3c30b5dc93)
Error:
if (view_name == permission.view_menu.name) and (permission_name == permission.permission.name):
AttributeError: 'NoneType' object has no attribute 'name'

caravel init logs -> https://gist.github.com/neevany/1a48fd4a5ce5c2621cc1b03fd85b2060

I think this error is happening because of the NULLS in below table. please help

<img width="511" alt="ab_permission_view" src="https://cloud.githubusercontent.com/assets/18303579/19636038/37ef3218-99e4-11e6-86cb-2eda60c62963.png">

**Comment by xrmx** [comment]
This is a dup of #1248, closing the other as this have more info.

**Comment by alanmcruickshank** [comment]
Hi Guys, I got this problem after upgrading an old caravel deployment (0.12.0) up to most recent superset (0.15.0).

If anyone else gets this - it is indeed due to the null values as above.

In my case there was only one: that for the permission "Superset". I had to find the null entry in `ab_permission_view`, find the `id`, then remove any entries in the table `ab_permission_view_role` which referenced it *and* then remove the `ab_permission_view` entry. Everything else now works fine.
